### PR TITLE
View API enhancements

### PR DESCRIPTION
**Pull request by NoahNsimbe**
**_WHAT DOES THIS PR DO?_**

- [x] Adds start and end date time filters to the app insights API.
- [x] Updates the test cases to include documentation for the new query parameters.
- [x] Removes unnecessary function used on the tasks utility file.
- [x] Enables compression of the response to improve performance

**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**
[MOB-7](https://airqoteam.atlassian.net/browse/MOB-7)

**_HOW DO I TEST OUT THIS PR?_**
Follow instructions in the readme file found [here](https://github.com/airqo-platform/AirQo-api/blob/view-api-enhancements/src/view/README.md#using-command-line)

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 
Follow instructions in the readme to view the API docs.

### CODE REVIEW

File: src/view/src/main/java/airqo/models/Insight.java [hunk 1]
```diff
@@ -43,6 +45,20 @@
...
+	@DateTimeFormat(pattern = dateTimeFormat)
+	@JsonIgnore
+	private Date startDateTime;
+
+	@Transient
+	@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = dateTimeFormat, timezone = "UTC")
+	@Getter(value = AccessLevel.PRIVATE)
+	@DateTimeFormat(pattern = dateTimeFormat)
+	@JsonIgnore
+	private Date endDateTime;
```
Comment (NoahNsimbe) [inline src/view/src/main/java/airqo/models/Insight.java:60]: Added to fields, `startDateTime` and `endDateTime` to use while querying start and end time resp.

File: src/view/src/main/java/airqo/ViewApplication.java [hunk 2]
```diff
@@ -3,9 +3,19 @@
...
+import java.util.TimeZone;
+
 @SpringBootApplication
 public class ViewApplication {
 
+	@PostConstruct
+	public void init(){
+		TimeZone.setDefault(TimeZone.getTimeZone("UTC"));
+		System.out.println("Timezone set to UTC : " + new Date());
+	}
```
Comment (NoahNsimbe) [inline src/view/src/main/java/airqo/ViewApplication.java:17]: Setting default timezone to UTC

File: src/view/src/test/java/airqo/MeasurementControllerTests.java [hunk 3]
```diff
@@ -204,33 +207,38 @@ public void shouldGenerateAppInsightsAPIDocs() throws Exception {
...
 
 		when(measurementService.apiGetInsights(predicate)).thenReturn(insights);
 
 		this.mockMvc.perform(get("/api/v1/view/measurements/app/insights")
 				.contextPath("/api/v1/view")
 				.header("Authorization", "Token my-jwt-token")
-				.param("siteId", "site-01,site-02")
+				.param("siteId", siteIds)
+				.param("startDateTime", simpleDateFormat.format(startDateTime))
+				.param("endDateTime", simpleDateFormat.format(endDateTime))
```
Comment (NoahNsimbe) [inline src/view/src/test/java/airqo/MeasurementControllerTests.java:233]: Testing `startDateTime` and `endDateTime` query parameters

**Review by 123MwanjeMike — APPROVED** [review]
Thanks @NoahNsimbe 
LGTM

**Review by Baalmart — APPROVED** [review]
LGTM

**Pull request merged**